src/sketch-app/sketch-app.html

```diff
@@ -1477,21 +1482,7 @@ <h2>Etch-a-Sketch</h2>
         }
 
         navigator.clipboard.readText()
-          .then(text => {
-            text = text.replace(/^[ '"]+|[ '"]+$/g, '');
-
-            const pattern = /^[\s0-9a-zA-Z]*$/;
-            var isValidData = pattern.test(text);
-            if(!isValidData) {
-              return this._alert('invalid clipboard input');
-            };
-
-            this.$.sketch.stop();
-            this.$.sketch.data = text;
-            this.peekID = null;
-            this.sketchDisabled = false;
-            this._alert('pasted from clipboard');
-          })
+          .then(this._pasteProto.bind(this))
```

Comment:
Hmm, I'm not sure if it should _just_ be this hunk that's moved to a new method, cuz some other stuff would be left out like the return cases...

Reply:
Oh! You could also turn that `.then` statement into
```js
.then(text => this._pasteProto(text))
```
just to be consistent with that `.catch` right below it